src/main/scala/it/pagopa/interop/backendforfrontend/service/types/PurposeProcessServiceTypes.scala

```
@@ -48,7 +59,84 @@ object PurposeProcessServiceTypes {
+        riskAnalysisDocument = pv.riskAnalysis.get.toApi,
+        createdAt = pv.createdAt,
+        firstActivationAt = pv.firstActivationAt.get
+      )
+    }
+  }
+
+  implicit class PurposeVersionsConverter(private val pv: PurposeProcess.PurposeVersion) extends AnyVal {
+    def toApi: PurposeVersion = PurposeVersion(
+      id = pv.id,
+      state = pv.state.toApi,
+      createdAt = pv.createdAt,
+      updatedAt = pv.updatedAt,
+      firstActivationAt = pv.firstActivationAt,
+      dailyCalls = pv.dailyCalls,
+      riskAnalysisDocument = pv.riskAnalysis.map(_.toApi)
+    )
+  }
+
+  implicit class ClientConverter(private val c: AuthorizationManagement.Client) extends AnyVal {
+    def toApi(hasKeys: Boolean): Client = Client(id = c.id, name = c.name, hasKeys = hasKeys)
+  }
+
+  implicit class RiskAnalysisDocumentConverter(private val rad: PurposeProcess.PurposeVersionDocument) extends AnyVal {
+    def toApi: PurposeVersionDocument =
+      PurposeVersionDocument(id = rad.id, contentType = rad.contentType, createdAt = rad.createdAt)
+  }
+
   implicit class PurposeConverter(private val p: PurposeProcess.Purpose) extends AnyVal {
+    def toApi(
+      eService: CatalogProcess.EService,
+      agreement: AgreementProcess.Agreement,
+      currentDescriptor: CatalogProcess.EServiceDescriptor,
+      currentVersion: Option[PurposeProcess.PurposeVersion],
+      producer: TenantProcess.Tenant,
+      consumer: TenantProcess.Tenant,
+      clients: Seq[AuthorizationManagement.Client],
+      hasKeys: Boolean,
+      waitingForApprovalVersion: Option[PurposeProcess.PurposeVersion]
+    ): Purpose = Purpose(
+      id = p.id,
+      title = p.title,
+      description = p.description,
+      consumer = CompactOrganization(id = consumer.id, name = consumer.name),
+      riskAnalysisForm =
+        p.riskAnalysisForm.getOrElse(PurposeProcess.RiskAnalysisForm(version = "v1", answers = Map.empty)).toApi,
```

Defaulting to `v1` can be misleading, moreover it does not exist in allowed versions and can create issues if sent back.
IMO this should be empty if missing, but let's check with the frontend if it's feasible